lib/puma/sd_notify.rb

```diff
@@ -137,10 +137,7 @@ def self.notify(state, unset_env=false)
       ENV.delete("NOTIFY_SOCKET") if unset_env
 
       begin
-        Addrinfo.unix(sock, :DGRAM).connect do |s|
-          s.close_on_exec = true
```

Comment:
I assume we remove this line because (from https://docs.ruby-lang.org/en/3.3/IO.html#method-i-close_on_exec-3D)

> Ruby sets close-on-exec flags of all file descriptors by default since Ruby 2.0.0. So you don’t need to set by yourself.